### ISSUE TITLE
SFTP Not Listing Remote Directory

### ISSUE THREAD

**Issue by topwebs** [Equals182/eqFTP#140]
Recently, we updated our server from Centos 5 to 6. After the update, FTP works normally with eqFTP, but we have an issue with SFTP. If you leave the Remote Path blank, SFTP works. However, if you put public_html into the Remote Path with any combination of slashes or none (which used to work), you see the remote directory listing flash up and then disappear. The error is:
```
 [node-error 4:20:38 PM] [eqFTP-ftpDomain]: Can't get in directory: public_html. Trying to get in /. The error is: stdin: is not a tty
NodeDebugUtils.js:114 handleLogEvent/utils/EventDispatcher.js:222 trigger/utils/EventDispatcher.js:260 triggerWithArray/utils/NodeConnection.js:479 NodeConnection._receive
NodeDebugUtils.js:114 [node-error 4:20:38 PM] [eqFTP-ftpDomain]: [q.p] Can't connect to server on connectionID: 2. Queue cleared now.
```

Brackets 1.3, eqFTP latest beta or 0.70. Any ideas of how to get Remote Path working?

**Comment by Equals182** [comment]
Hi

This is something new, I think. Don't remember this error `stdin: is not a tty`
I'll see if I can do something with it. 

Is it flash up with contents of `public_html`? Just like it works fine first but after that something goes wrong and it disappears?

**Comment by topwebs** [comment]
For some reason, I can't now reproduce the content flash. I tried a lot of combinations of slashes and List vs MLSD commands and different versions, so not sure which one gave the flash of content. I am now using the latest beta. Upon trying to connect, I see the Connect notification, immediately followed by the Disconnect notification. The screen remains blank.

**Comment by Equals182** [comment]
And the error is still same, I assume?
I'm sory for that inconvinience. I'll try to fix it somehow.

**Comment by topwebs** [comment]
Yes, the error is the same. I might add that SFTP with WinSCP works normally with the new server. A WinSCP page says:
Technical Background of Terminal Limitations
WinSCP does not support commands that require terminal emulation or user input.
Terminal emulation cannot be implemented at all as it is not possible to combine terminal emulation and file transfers in the same session. The reason is that with terminal emulation control characters have special meaning that is interpreted by the server. On the other hand the same characters (bytes) have special meaning in the file transfer protocols or may be simply present in the (binary) files being transferred.
If you happen to try execute command requiring terminal emulation, it will typically refuse to run with error message like “Error opening terminal”, “No TTY available”, “stdin: is not a tty” or “Couldn’t open /dev/tty for reading”.
WinSCP may possibly support user input. However most commands requiring user input does that via terminal emulation features. If your command needs plain input without terminal emulation you can use input redirection like:
echo "some input" | command

**Comment by Equals182** [comment]
Oh that's a useful information, thank you!

**Comment by Equals182** [comment]
Alright, I see where the error is. It wasn't so hard to find and I have some ideas on how to fix it.

**Long story:**
I wrote this stuff by myself. I was thinking about how to check if folder exists, like, for sure. So I came to idea of `cwd + pwd`. Like, going to directory and checking if current directory is same as expected. But apparentely SFTP is so different and there's no such thing like cwd and pwd by default. But then I made a workaround with SSH (or SCP, I don't know). It sends `cd + pwd` via console. I knew it's not good solution, but I had nothing else on my mind.
Now I need 100% working alternative for `cd + pwd` thing OR really different alternative for checking if folder exists.
You may ask, WTH, why did I did this cwdpwd shit in first place. Well, apparentely not all FTP servers can handle proper response codes, like giving me 200 code on unexistent folder (there was an issue for that). So I made this, assuming there's no *SO* dumb server, that would return me unexistent directory on `pwd` command. And then it got its way to SFTP. And now it needs to be fixed.

**TL;DR;**
I will look for alternative for checking folders :)

**Comment by Equals182** [comment]
Hello, sir
I'm glad to announce a patch, that may help you resolve your issue. To make it work, please follow to **Advanced** tab in **Connection**'s settings and set `Use SSH commands with SFTP` to `No`. 

Please, keep in mind, that there's no way to automatically correct `Remote Path` when this setting set to `No`, since eqFTP will not be able to send `PWD` command to get current working directory and set it as remote root path. This may cause some unexpected issues when path set to some unexistent folder on server.

I would also recommend trying `./` combination. I've discovered that `.` leads to SFTP root folder (for current account). So you may try setting it to `./public_html` if nothing helps.

I tried to check everything that somehow touches those SSH commands, but I may miss something, so if you still experience problems, please provide a log.

**Comment by topwebs** [comment]
Setting SSH to No works with Remote Path set to public_html with or without slashes. Thanks.

**Comment by Equals182** [comment]
I guess we can close this issue then.
Thanks for using eqFTP! Have fun!
